### PR TITLE
🐛 read config referent from the desired config

### PR DESCRIPTION
**Pull request by qiujian16**
## Related issue(s)

Fixes #

### CODE REVIEW

File: pkg/utils/addon_config.go [hunk 1]
```diff
@@ -125,9 +125,10 @@ func GetAddOnConfigRef(
 	configReferences []addonapiv1alpha1.ConfigReference,
```
Comment (haoqing0110) [inline pkg/utils/addon_config.go:127]: > // GetAddOnConfigRef returns the first addon config ref for the given config type


Need to update the func description as well.

File: pkg/utils/addon_config.go [hunk 2]
```diff
@@ -120,14 +120,15 @@ func GetAddOnDeploymentConfigSpecHash(config *addonapiv1alpha1.AddOnDeploymentCo
...
-// GetAddOnConfigRef returns the first addon config ref for the given config type
+// GetAddOnConfigRef returns the last addon config ref for the given config type
 func GetAddOnConfigRef(
 	configReferences []addonapiv1alpha1.ConfigReference,
 	group, resource string) (bool, addonapiv1alpha1.ConfigReference) {
 
-	for _, config := range configReferences {
-		if config.Group == group && config.Resource == resource {
-			return true, config
+	// iterate the config reversely, the one with larger index is picked at first
```
Comment (zhujian7) [inline pkg/utils/addon_config.go]: Have we already sorted this in another controller(addonconfiguration controller, with a graph structure)? I need to confirm this.

Comment (qiujian16) [reply pkg/utils/addon_config.go]: this is use in the helm template addon to get values, which will need this to read in reverse order.

**Pull request merged**